### ISSUE TITLE
Support newer sftp protocol versions

### ISSUE THREAD

**Issue by tionis** [pkg/sftp#593]
Even though v3 is the most widely supported one, the most up-to-date specification is v6: https://www.sftp.net/specification

**Comment by puellanivis** [comment]
Yes, we are aware of the newer versions, example: https://github.com/pkg/sftp/blob/master/internal/encoding/ssh/filexfer/fx.go#L60-L61

The problem is that basically no one supports the newer versions, particularly not even OpenSSH: https://github.com/openssh/openssh-portable/blob/13cc78d016b67a74a67f1c97c7c348084cd9212c/sftp.h#L86-L96

Reading this section of the Wikipedia page is pretty informative: https://en.wikipedia.org/wiki/SSH_File_Transfer_Protocol#History_and_development 

We’re considering in v2 of the API how we could provide a better ability for people to implement later versions if they need it.
